src/features/counterfactual/hooks/usePendingSafeStatuses.ts

```diff
@@ -85,6 +85,9 @@ const usePendingSafeStatus = (): void => {
     if (!provider || !safeAddress) return
 
     const checkDeploymentStatus = async () => {
+      // In case the safe info hasn't been updated yet when switching safes
+      if ((await provider.getNetwork()).chainId !== BigInt(safe.chainId)) return
```

Comment:
Nit pick: this is a bit hard to read.

```suggestion
      const { chainId } = await provider.getNetwork()
      if (chainId !== BigInt(safe.chainId)) return
```